test/unit/legacy_wrappers/gridfs.test.js

```diff
@@ -48,4 +48,52 @@ describe('legacy_wrappers/gridfs.js', () => {
       LegacyGridFSBucketWriteStream
     );
   });
+
+  context('delete', function () {
+    it('correctly handles parameters when options are provided', function () {
+      const spy = sinon.spy(mongodbDriver.GridFSBucket.prototype, 'delete');
+      const opts = { timeoutMS: 10 };
+      const oid = new mongodbDriver.ObjectId();
+      bucket.delete(oid, opts, () => {});
+      expect(spy).to.be.calledWithExactly(oid, opts);
```

Comment:
Can we see what happens when no `callback` function is provided, for each of the operations? I'm assuming `callback` is optional from the type checking in the functions.

Reply:
Whether or not callback is provided doesn't affect what gets called by the actual driver's GridFS implementation. This library wraps the driver to allow legacy applications that use callback syntax instead of async/await or promises to use the driver without having to refactor their entire codebases.

Reply:
Thanks for the context! My original thinking is that we're type checking for whether or not a callback is provided and so might as well test it. I can leave this comment thread open and move this to team review, to see if anyone else has thoughts.

Reply:
We should be updating `test/tools/api.js` to cover what Aditi is talking about, we use  that big descriptor object to enumerate all possible invocation permutations. You can reference: https://github.com/mongodb-js/nodejs-mongodb-legacy/pull/45/files#diff-632f19f0e90a8ffb3ff08c63237c1c52ab5d57cf6b592ad5e31e57c78d8fbe7c and check the output of the tests to see the new rename/drop/delete coverage